code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/interactions/UIEvent.kt

```diff
@@ -0,0 +1,64 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui.interactions
+
+import com.adobe.marketing.mobile.aepcomposeui.aepui.AepUI
+import com.adobe.marketing.mobile.aepcomposeui.aepui.state.AepUIState
+import com.adobe.marketing.mobile.aepuitemplates.AepUITemplate
+
+/**
+ * Sealed interface representing different types of UI events in AEP.
```

Comment:
nit: `..representing different types of UI events that can be produced by supported UI templates`